js/main.js

```diff
@@ -378,6 +378,33 @@ function watchlistFilms() {
     $removeBtn.textContent = 'Remove';
     $watchlistInfo.append($removeBtn);
 
+    const $removeBtns = document.querySelectorAll('.remove-watchlist');
+    for (let i = 0; i < $removeBtns.length; i++) {
+      const $btn = $removeBtns[i];
+      $btn.addEventListener('click', function () {
+        const $card = $btn.parentElement;
+        const cardId = $card.getAttribute('data-watch-id');
+        for (let j = 0; j < data.watchlist.length; j++) {
+          const dataArr = data.watchlist[j];
+          if (dataArr.id === cardId) {
+            data.watchlist.splice(j, 1);
+          }
+        }
+        removeWatchlist(cardId);
+        toggleWatchNoEntry();
+      });
+    }
+  }
```

Comment:
Since this is inside of your render function, what's the purpose of the loop? Aren't you only creating 1 button? Why not just add it to that button rather than loop everytime this is called?